earsketch/earsketch-webclient
# Language Detection

Language detection will only happen the first time a web browser navigates to EarSketch. If there is no `locale` set in local storage, attempt to detect the language from the web browser's `navigator` language settings. Since users will likely have a region specific dialect selected (ex. `en-US` or `en-gb`), we also check the parent language (to the left of the hyphen) to see if we support that.

The included test attempts to cover all of the possible formats expected from different browsers.

This also refactors the `Curriculum` component to automatically re-fetch the curriculum locale json data files (TOC, search, pages) so we can remove imports of `curriculumState` from `Bubble` and `LocaleSelector` for simpler testing.

Resolves GTCMT/earsketch#2617

### Testing steps:

1. Delete your `persist:app` key from `localstorage` for `https://earsketch-test.ersktch.gatech.edu`
2. Set your browser's language to one of our supported languages (French, Spanish, etc)
3. Refresh / load this PR build: https://earsketch-test.ersktch.gatech.edu/pr-360/

The app should default to the language you set in your browser. If you chose an unsupported language then it should default to English.
